include/dlaf/eigensolver/reduction_to_band/impl.h

```diff
@@ -36,15 +39,28 @@
 #include "dlaf/matrix/panel.h"
 #include "dlaf/matrix/tile.h"
 #include "dlaf/matrix/views.h"
+#include "dlaf/schedulers.h"
 #include "dlaf/sender/keep_future.h"
 #include "dlaf/sender/traits.h"
+#include "dlaf/types.h"
+#include "dlaf/util_math.h"
 #include "dlaf/util_matrix.h"
 
 #include "dlaf/eigensolver/reduction_to_band/api.h"
 #include "dlaf/factorization/qr.h"
 
 namespace dlaf::eigensolver::internal {
 
+// Given a vector of vectors, reduce all vectors in the first one using sum operation
+template <class T>
+void reduceColumnVectors(std::vector<common::internal::vector<T>>& battery) {
```

Comment:
What does `battery` mean in this context?

Reply:
So how about just `columns`? Or `columnVectors`? Or `vectors`? Or more something more specific with a suffix of `columns` or `vectors` if it makes sense for the algorithm. The function is called `reduceColumnVectors` after all.

Reply:
Regarding using `Matrix`: is it a ragged vector of vectors, i.e. are the vectors different lengths

Reply:
I guess it's not a ragged vector if they're reduced to the first vector... This is side-tracking, but what keeps you from using a `Tile` or a `Matrix` here?

Reply:
IIRC it was about when the memory was allocated.

With a `just(vector<vector<>>)` I can decide at runtime how much but more importantly to when to actually allocate, i.e. just at runtime and not at schedule-time.

Currently, with `Matrix`/`Tile` we can just pre-allocate at schedule-time, and it might consume a lot of memory even when just a single buffer can be re-used thanks to memory managers (and not manually, like with Panel).

I would just go for renaming, but in case @msimberg or @rasolca you have a better idea, I'm open to do it now or in a next PR 😉

Any preference for the name? `columnVectors`?

Reply:
I'd have to look closer to be able to comment about the `vector` vs `Tile` issue, but as you said that can be done separately.

I'd be happy with `columnVectors`. I think it describes what it is better than `battery`.